pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -555,7 +577,17 @@ func (gce *GCECloud) EnsureLoadBalancer(name, region string, requestedIP net.IP,
 	// Deal with the firewall next. The reason we do this here rather than last
 	// is because the forwarding rule is used as the indicator that the load
 	// balancer is fully created - it's what getLoadBalancer checks for.
-	firewallExists, firewallNeedsUpdate, err := gce.firewallNeedsUpdate(name, region, ipAddress, ports)
+	// Check if user specified the allow source range
+	sourceRanges := []string{defaultLBSourceRange}
+	val, ok := annotations[gceLBAllowSourceRange]
```

Comment:
I meant retrieve the annotation via a method on the type, so everyone who uses it insn't reaching into the map on their own. That way if we made it a field on the svc instead of an annotation only one piece of code will have to change. Eg: https://github.com/kubernetes/contrib/blob/master/service-loadbalancer/service_loadbalancer.go#L222